siatest/testgroup.go

```diff
@@ -267,13 +274,36 @@ func mapToSlice(m map[*TestNode]struct{}) []*TestNode {
 	return tns
 }
 
-// randomDir is a helper functions that returns a random directory path
-func randomDir() string {
-	dir, err := TestDir(strconv.Itoa(fastrand.Intn(math.MaxInt32)))
-	if err != nil {
-		panic(errors.AddContext(err, "failed to create testing directory"))
+// randomNodeDir generates a random directory for the provided node params if
+// Dir wasn't set using the provided parentDir and a randomized suffix.
+func randomNodeDir(parentDir string, nodeParams *node.NodeParams) {
+	if nodeParams.Dir != "" {
+		return
+	}
+	nodeDir := ""
+	if nodeParams.Gateway != nil || nodeParams.CreateGateway {
+		nodeDir += "g"
+	}
+	if nodeParams.ConsensusSet != nil || nodeParams.CreateConsensusSet {
+		nodeDir += "c"
+	}
+	if nodeParams.TransactionPool != nil || nodeParams.CreateTransactionPool {
+		nodeDir += "t"
 	}
-	return dir
+	if nodeParams.Wallet != nil || nodeParams.CreateWallet {
+		nodeDir += "w"
+	}
+	if nodeParams.Renter != nil || nodeParams.CreateRenter {
+		nodeDir += "r"
+	}
+	if nodeParams.Host != nil || nodeParams.CreateHost {
+		nodeDir += "h"
+	}
+	if nodeParams.Miner != nil || nodeParams.CreateMiner {
+		nodeDir += "m"
+	}
+	nodeDir += fmt.Sprintf("-%s", hex.EncodeToString(fastrand.Bytes(4)))
```

Comment:
any reason that you are doing this over using the `RandomSufffix()` function?

Reply:
Good point.